clang/lib/Sema/Sema.cpp

```diff
@@ -1962,13 +1962,14 @@ Sema::targetDiag(SourceLocation Loc, unsigned DiagID, FunctionDecl *FD) {
   if (LangOpts.OpenMP)
     return LangOpts.OpenMPIsDevice ? diagIfOpenMPDeviceCode(Loc, DiagID, FD)
                                    : diagIfOpenMPHostCode(Loc, DiagID, FD);
-  if (getLangOpts().CUDA)
-    return getLangOpts().CUDAIsDevice ? CUDADiagIfDeviceCode(Loc, DiagID)
-                                      : CUDADiagIfHostCode(Loc, DiagID);
 
   if (getLangOpts().SYCLIsDevice)
     return SYCLDiagIfDeviceCode(Loc, DiagID);
 
+  if (getLangOpts().CUDA)
```

Comment:
Can you explain why this was required?

Reply:
This is required in order to perform SYCL device diagnosis in case of **device-sycl** compilation. Otherwise, the CUDA host one is wrongly picked.